Source/WebCore/platform/graphics/GraphicsLayer.h

```diff
@@ -524,6 +535,7 @@ class GraphicsLayer : public RefCounted<GraphicsLayer> {
     virtual void setContentsToSolidColor(const Color&) { }
     virtual void setContentsToPlatformLayer(PlatformLayer*, ContentsLayerPurpose) { }
     virtual void setContentsToPlatformLayerHost(LayerHostingContextIdentifier) { }
+    virtual void setContentsToVideoElement(HTMLVideoElement&, ContentsLayerPurpose) { }
```

Comment:
In order to avoid referencing HTML element classes inside the graphics layer class, you could possibly create a new, more generic class to use here instead of HTMLVideoElement. This class would then declare the virtual methods needed, and HTMLVideElement could subclass and override the methods. However, I don't think this is needed to move forward with the current patch, and could be done in a follow-up patch, if you think it makes sense.

Reply:
GraphicsLayer cannot know about HMTL elements.